app/src/main/java/ch/epfl/sdp/HistoryFirestoreInteractor.java

```diff
@@ -1,42 +1,35 @@
 package ch.epfl.sdp;
 
-import androidx.annotation.VisibleForTesting;
-
-import com.google.android.gms.tasks.OnFailureListener;
-import com.google.android.gms.tasks.OnSuccessListener;
-
+import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.CompletableFuture;
 
 import ch.epfl.sdp.firestore.ConcreteFirestoreInteractor;
-import ch.epfl.sdp.firestore.FirestoreInteractor;
-import ch.epfl.sdp.firestore.QueryHandler;
 
-public class HistoryFirestoreInteractor {
+public class HistoryFirestoreInteractor extends ConcreteFirestoreInteractor {
 
-    private FirestoreInteractor fsi;
     private Account user;
 
     HistoryFirestoreInteractor(Account user) {
-        this.fsi = new ConcreteFirestoreInteractor();
+        super();
         this.user = user;
     }
 
-    public void read(QueryHandler handler) {
+    public CompletableFuture<Map<String, Map<String, Object>>> readHistory() {
         String path = "History/" + user.getId() + "/Positions";
-        fsi.readCollection(path, handler);
+        return readCollection(collectionReference(path));
     }
 
 
-    public void write(Map<String, Object> content, OnSuccessListener success,
-                      OnFailureListener failure) {
+    public CompletableFuture<Void> write(Map<String, Object> content) {
         String path = "History/" + user.getId() + "/Positions";
```

Comment:
Would be nice to move the path resolution to a separate function to reuse it or change it later :-)

Reply:
Good point, I'm going to do that